### PR TITLE
Arif khalid transaction list commands

### LINKED ISSUE #85: Print All Transactions command
`trans` should display all their transactions to the user in reverse chronological order. i.e. most recent displayed above less recent.

### LINKED ISSUE #86: Search Transactions by currency
`trans c <CURRENCY>` should display all the transactions of that currency account

### LINKED ISSUE #87: Search transactions by date
Contingent on a LocalDateTime field in Transaction that allows the transaction manager to search by date.
`trans d <DD-MM-YYYY>` should return all transactions of that date

### LINKED ISSUE #88: Search transaction by description
`trans desc <Description to search for>` should return all transactions with descriptions matching the description to search for.

### LINKED ISSUE #89: Search for expenditure within a month
`trans m <MM-YYYY>` should print all the users expenditure within the month passed in by the user

### PR DESCRIPTION
**Pull request by Arif-Khalid**
# Description

## Type of Change

- [ ] Possible breaking change (Change in feature that may cause existing features to not work correctly, e.g. change in
  schema)
- [x] Non breaking change (Bug fix, New feature)

## What type of PR is this?

- [ ] bug fix
- [ ] cleanup
- [ ] documentation
- [x] feature

## What this PR does / Why we need it:

```docs
Implements core features of printing and searching through transactions made by the user
```

## Which issue(s) this PR fixes, if:
closes #85
closes #86 
closes #87 
closes #88 
closes #89 

#### Additional documentation e.g., usage docs, testing docs etc.:

```docs

```

## Checklist

- [x] Changes have been tested
- [x] All automated tests pass
- [x] Relevant Tags and Milestones have been added

### CODE REVIEW

File: src/main/java/seedu/duke/Account.java [hunk 1]
```diff
@@ -50,13 +50,9 @@ public void updateBalance (BigDecimal changeInBalance, String action) throws Not
...
     }
 
-    public static String currencyToString (Currency currency) {
-        return currency.name();
-    }
-
     @Override
     public String toString () {
-        String currencyType = currencyToString(currency);
+        String currencyType = currency.name();
```
Comment (Gan868611) [inline src/main/java/seedu/duke/Account.java:55]: Nice work on removing redundant method

File: src/main/java/seedu/duke/commands/TransactionCommand.java [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+            InvalidSearchTransactionByCurrencyException, NoTransactionsOfSearchParameterException,
+            InvalidTransactionFlagException, IllegalArgumentException {
+        TransactionManager transactionManager = TransactionManager.getInstance();
+        String transactionsString;
+        switch (flag) {
+        case TransactionFlag.TRANSACTION_DESC_FLAG:
+            if (args.length < 3) {
+                throw new InvalidSearchTransactionByDescException();
+            }
+            String description = findSubstringToEndAfterASubstring(input, TransactionFlag.TRANSACTION_DESC_FLAG);
```
Comment (Gan868611) [inline src/main/java/seedu/duke/commands/TransactionCommand.java:123]: i think this suggestion does the same thing too.  Just FYI, current method has no problem. 
```suggestion
            String description = input.split("/(TransactionFlag.TRANSACTION_DESC_FLAG)", 2 )[1].trim();
```

File: src/main/java/seedu/duke/commands/TransactionCommand.java [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+    /**
+     * Method that searches a string for a substring
+     * Upon finding that substring, returns the rest of the string after that substring,
+     * excluding the searched substring
+     *
+     * @param stringToSearch The string to search within
+     * @param substring      The substring to search for
+     * @return The rest of the input string after and excluding the input substring, null if substring is not found
+     */
+    private String findSubstringToEndAfterASubstring (String stringToSearch, String substring) {
```
Comment (Bawfen) [inline src/main/java/seedu/duke/commands/TransactionCommand.java:87]: Perhaps we could rename this function to `substringAfterKeyword` so its purpose is a bit clearer?

File: src/main/java/seedu/duke/commands/TransactionCommand.java (hunk 2 above)
Comment (Bawfen) [reply src/main/java/seedu/duke/commands/TransactionCommand.java:123]: Good suggestion, though I think the current implementation abstracts it a bit more cleanly

File: src/main/java/seedu/duke/commands/TransactionCommand.java (hunk 3 above)
Comment (Arif-Khalid) [reply src/main/java/seedu/duke/commands/TransactionCommand.java]: Method renamed to substringAfterKeyword

**Pull request merged**